mijia/src/lib.rs

```diff
@@ -1,137 +1,107 @@
-use blurz::{
-    BluetoothAdapter, BluetoothDevice, BluetoothDiscoverySession, BluetoothGATTCharacteristic,
-    BluetoothSession,
-};
+use bluez_generated::generated::OrgBluezGattCharacteristic1;
+use dbus::arg::RefArg;
+use dbus::nonblock::stdintf::org_freedesktop_dbus::ObjectManager;
+use itertools::Itertools;
 use std::cmp::max;
 use std::collections::HashMap;
 use std::convert::TryInto;
-use std::error::Error;
 use std::fmt::{Display, Formatter};
 use std::fs::File;
 use std::io::{self, BufRead, BufReader, ErrorKind};
-use std::thread;
 use std::time::Duration;
 
-const SCAN_DURATION: Duration = Duration::from_millis(5000);
-const CONNECT_TIMEOUT_MS: i32 = 10_000;
+pub mod session;
+pub use session::{MijiaEvent, MijiaSession};
 
-const MIJIA_SERVICE_DATA_UUID: &str = "0000fe95-0000-1000-8000-00805f9b34fb";
+pub const MIJIA_SERVICE_DATA_UUID: &str = "0000fe95-0000-1000-8000-00805f9b34fb";
 pub const SERVICE_CHARACTERISTIC_PATH: &str = "/service0021/char0035";
-const CONNECTION_INTERVAL_CHARACTERISTIC_PATH: &str = "/service0021/char0045";
+pub const CONNECTION_INTERVAL_CHARACTERISTIC_PATH: &str = "/service0021/char0045";
 /// 500 in little-endian
-const CONNECTION_INTERVAL_500_MS: [u8; 3] = [0xF4, 0x01, 0x00];
+pub const CONNECTION_INTERVAL_500_MS: [u8; 3] = [0xF4, 0x01, 0x00];
```

Comment:
Why do these constants all need to be pub?

Reply:
they probably only needed to be pub while I was mid-refactor. Fixed now.